src/Appwrite/Network/Validator/DNS.php

```diff
@@ -2,115 +2,93 @@
 
 namespace Appwrite\Network\Validator;
 
-use Utopia\DNS\Client;
-use Utopia\DNS\Message;
-use Utopia\DNS\Message\Question;
+use Swoole\Coroutine\WaitGroup;
 use Utopia\DNS\Message\Record;
-use Utopia\Domains\Domain;
-use Utopia\System\System;
-use Utopia\Validator;
+use Utopia\DNS\Validator\DNS as BaseDNS;
 
-class DNS extends Validator
+class DNS extends BaseDNS
 {
-    public function __construct(
-        protected string $target,
-        protected int $type = Record::TYPE_CNAME,
-        protected string $server = ''
-    ) {
-        $this->server = $server ?: System::getEnv('_APP_DNS', '8.8.8.8');
-    }
-
-    public function getDescription(): string
+    /**
+     * @var array<string>
+     */
+    protected array $dnsServers = [];
+
+    /**
+     * @param string $target Expected value for the DNS record
+     * @param int $type Type of DNS record to validate
+     *  For value, use const from Record, such as Record::TYPE_A
+     *  When using CAA type, you can provide exact match, or just issuer domain as $target
+     * @param string|array<string> $dnsServers DNS server IP(s) or domain(s) to use for validation
+     */
+    public function __construct(string $target, int $type = Record::TYPE_CNAME, string|array $dnsServers = self::DEFAULT_DNS_SERVER)
     {
-        return 'Invalid DNS record.';
+        parent::__construct($target, $type, is_array($dnsServers) ? $dnsServers[0] : $dnsServers);
+
+        $this->dnsServers = is_array($dnsServers) ? $dnsServers : [$dnsServers];
     }
 
-    public function isValid($value): bool
+    /**
+     * Validate DNS record value against multiple DNS servers
+     *
+     * @param mixed $value
+     * @return bool
+     */
+    public function isValid(mixed $value): bool
     {
-        if (!is_string($value) || trim($value) === '') {
-            return false;
-        }
-
-        $client = new Client($this->server);
-        try {
-            $response = $client->query(Message::query(
-                new Question($value, $this->type)
-            ));
-        } catch (\Throwable) {
-            return false;
+        if (\count($this->dnsServers) === 1) {
+            return $this->isValidWithDNSServer($value, $this->dnsServers[0]);
         }
```

Comment:
We might not need this, technically, our coroutine approach should work fine and quick even for just 1 dns server